platform/ios/app/MBXViewController.m

```diff
@@ -908,6 +912,33 @@ - (void)styleDynamicPointCollection
     [self.mapView.style addLayer:layer];
 }
 
+- (void)styleCountryLabelLanguage
+{
+    NSString *bestLanguageForUser = [NSString stringWithFormat:@"{name_%@}", [self bestLanguageForUser]];
+    MGLSymbolStyleLayer *countryLayer = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"country-label-lg"];
+    MGLStyleConstantValue<NSString *> *countryLabel = (MGLStyleConstantValue<NSString *> *)countryLayer.textField;
+    NSString *language = [countryLabel.rawValue isEqual:bestLanguageForUser] ? @"{name}" : bestLanguageForUser;
+    countryLayer.textField = [MGLStyleValue<NSString *> valueWithRawValue:language];
+}
+
+- (NSString *)bestLanguageForUser
+{
+    NSArray *supportedLanguages = @[ @"en", @"es", @"fr", @"de", @"ru", @"zh" ];
+    NSArray<NSString *> *preferredLanguages = [NSLocale preferredLanguages];
+    NSString *bestLanguage;
+
+    for (NSString *language in preferredLanguages) {
+        NSString *thisLanguage = [language substringToIndex:2];
```

Comment:
What if the user’s preferred language is set to Aromanian (`rup`)? 😉 FYI, this would be more robust, but what you’ve got is fine for a demo:

```objc
[NSLocale localeWithIdentifier:language].languageCode
```